pandas/core/series.py

```diff
@@ -4303,6 +4308,94 @@ def between(self, left, right, inclusive=True) -> "Series":
 
         return lmask & rmask
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _convert_dtypes(
+        self: ABCSeries,
+        infer_objects: bool = True,
+        convert_string: bool = True,
+        convert_integer: bool = True,
+        convert_boolean: bool = True,
+    ) -> ABCSeries:
+        """
+        Convert objects to best possible type, and optionally,
+        columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        Parameters
+        ----------
+        infer_objects : bool, default True
+            Whether object dtypes should be converted to the best possible types.
+        convert_string : bool, default True
+            Whether object dtypes should be converted to ``StringDtype()``.
+        convert_integer : bool, default True
+            Whether, if possible, conversion can be done to integer extension types.
+        convert_boolean : bool, defaults True
+            Whether object dtypes should be converted to ``BooleanDtypes()``.
+
+        Returns
+        -------
+        Series
+            copy of Series with new (or existing) dtype
+        """
+        input_series = self
+        if infer_objects:
```

Comment:
so almost *all* of this logic needs to go in panda.core.dtypes.cast, right near / after maybe_convert_objects. these will take Union[ExtensionArray, PandasArray], e.g. pass `self._values` to them.

this is where inference happens in the pandas, not here as its too long already (these modules).

Reply:
done.